www/modules/recalls/controllers/SearchCtrl.js

```diff
@@ -65,21 +65,36 @@ main.controller('SearchCtrl', function (
      * @function
      * @param {string} barcode A UPC barcode string to search for.
      */
-    self.searchByBarcode = function(barcode) {
+    self.searchByBarcode = function (barcode) {
         //No need to continue if we don't have a barcode to work with.
-        if(!barcode) {
+        if (!barcode) {
             return;
         }
 
         $location.search({keywords: null, barcode: barcode});
 
         foodEnforcementService.getRecallsByBarcode(barcode)
-            .success(function(result) {
+            .success(function (result) {
                 self.recalls = result.results || [];
             })
-            .error(function() {
+            .error(function () {
                 self.recalls = [];
-                //TODO: can we get product info from barcode and do a keyword search?
+
+                // get product info from barcode and do a keyword search
+                factualUpcService.getData(barcode).success(function (result) {
+                    var products = factualUpcService.getProducts(result);
+                    var product;
+                    if (products && products.length) {
+                        product = products[0];
+                        foodEnforcementService.getRecallsByKeyword(product.name)
+                            .success(function (result) {
```

Comment:
There is no need for success and error handlers here, just simply call the getResultsByKeywords method